File: backend/src/main/java/backend/mulkkam/cup/domain/IntakeType.java
package backend.mulkkam.cup.domain;

import static backend.mulkkam.common.exception.errorCode.NotFoundErrorCode.NOT_FOUND_INTAKE_TYPE;

import backend.mulkkam.common.exception.CommonException;
import java.util.Arrays;

public enum IntakeType {

    WATER(1.0),
    COFFEE(0.95),
    ;

    private final double hydrationRatio;

    IntakeType(double hydrationRatio) {
        this.hydrationRatio = hydrationRatio;
    }

    public static IntakeType findByName(String name) {
        return Arrays
                .stream(values())
                .filter(intakeType -> intakeType.name().equalsIgnoreCase(name))
                .findFirst()
                .orElseThrow(() -> new CommonException(NOT_FOUND_INTAKE_TYPE));
    }

    public int calculateHydration(int intakeAmount) {
        return (int) Math.round(intakeAmount * hydrationRatio);
    }
}

File: android/app/src/main/java/com/mulkkam/domain/model/IntakeType.kt
package com.mulkkam.domain.model

enum class IntakeType {
    WATER,
    COFFEE,
    UNKNOWN,
    ;

    // TODO: 추후 서버에서 받아오는 IntakeType 반영
    fun toLabel(): String =
        when (this) {
            WATER -> "물"
            COFFEE -> "커피"
            UNKNOWN -> ""
        }

    fun toColorHex(): String =
        when (this) {
            WATER -> "#90E0EF"
            COFFEE -> "#C68760"
            UNKNOWN -> ""
        }

    companion object {
        fun from(intakeType: String): IntakeType =
            when (intakeType) {
                WATER.name -> WATER
                COFFEE.name -> COFFEE
                else -> UNKNOWN
            }
    }
}
